sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_EventProcessor.md

```diff
@@ -0,0 +1,230 @@
+    {
+        StorageContainer = storageContainer;
+    }
+
+    private const string OwnershipPrefixFormat = "{0}/{1}/{2}/ownership/";
+    private const string OwnerIdentifierMetadataKey = "ownerid";
+
+    // Ownership information is stored as metadata on blobs in Azure Storage. To list ownership information we list all the blobs in blob storage (with their coresponding metadata) and then
+    // extract the ownership information from the metadata.
+    protected override async Task<IEnumerable<EventProcessorPartitionOwnership>> ListOwnershipAsync(CancellationToken cancellationToken = default)
+    {
+        List<EventProcessorPartitionOwnership> partitonOwnerships = new List<EventProcessorPartitionOwnership>();
+        string ownershipBlobsPefix = string.Format(OwnershipPrefixFormat, FullyQualifiedNamespace.ToLowerInvariant(), EventHubName.ToLowerInvariant(), ConsumerGroup.ToLowerInvariant());
+
+        await foreach (BlobItem blob in StorageContainer.GetBlobsAsync(traits: BlobTraits.Metadata, prefix: ownershipBlobsPefix, cancellationToken: cancellationToken).ConfigureAwait(false))
+        {
+            partitonOwnerships.Add(new EventProcessorPartitionOwnership()
+            {
+                ConsumerGroup = ConsumerGroup,
+                EventHubName = EventHubName,
+                FullyQualifiedNamespace = FullyQualifiedNamespace,
+                LastModifiedTime = blob.Properties.LastModified.GetValueOrDefault(),
+                OwnerIdentifier = blob.Metadata[OwnerIdentifierMetadataKey],
+                PartitionId = blob.Name.Substring(ownershipBlobsPefix.Length),
+                Version = blob.Properties.ETag.ToString()
+            });
+        }
+
+        return partitonOwnerships;
+    }
+
+    // To claim ownership of a partition, we have to write a blob to blob storage (if this partition has never been claimed before) or update the metadata of an existing blob.
+    protected override async Task<IEnumerable<EventProcessorPartitionOwnership>> ClaimOwnershipAsync(IEnumerable<EventProcessorPartitionOwnership> desiredOwnership, CancellationToken cancellationToken = default)
+    {
+        List<EventProcessorPartitionOwnership> claimedOwnerships = new List<EventProcessorPartitionOwnership>();
+    
+        foreach (EventProcessorPartitionOwnership ownership in desiredOwnership)
+        {
+            Dictionary<string, string> ownershipMetadata = new Dictionary<string, string>()
+            {
+                { OwnerIdentifierMetadataKey, ownership.OwnerIdentifier },
+            };
+
+            // Construct the path to the blob and get a blob client for it so we can interact with it.
+            string ownershipBlob = string.Format(OwnershipPrefixFormat + ownership.PartitionId, ownership.FullyQualifiedNamespace.ToLowerInvariant(), ownership.EventHubName.ToLowerInvariant(), ownership.ConsumerGroup.ToLowerInvariant());
```

Comment:
nit:  seems odd to have a string concatenation embedded in a `string.Format` call.